src/transformers/tokenization_utils.py

```diff
@@ -49,6 +49,126 @@
 TOKENIZER_CONFIG_FILE = "tokenizer_config.json"
 
 
+class Trie:
+    """
+    Trie in Python. Creates a Trie out of a list of words. The trie is used to split on `added_tokens` in one pass
+    """
+
+    def __init__(self):
+        self.data = {}
+
+    def add(self, word: str):
+        """
+        Passes over every char (utf-8 char) on word and recursively adds it to the internal `data` trie representation.
+        The special key `""` is used to represent termination.
+
+        This function is idempotent, adding twice the same word will leave the trie unchanged
+
+        >>> trie = Trie()
+
+        >>> trie.add("Hello 友達")
+
+        >>> trie.data
+
+        {"H": {"e": {"l": {"l": {"o": {" ": {"友": {"達": {"": 1}}}}}}}}}
+
+        >>> trie.add("Hello")
+
+        >>> trie.data
+
+        {"H": {"e": {"l": {"l": {"o": {"": 1, " ": {"友": {"達": {"": 1}}}}}}}}}
+        """
+        if not word:
+            # Prevent empty string
+            return
+        ref = self.data
+        for char in word:
+            ref[char] = char in ref and ref[char] or {}
+            ref = ref[char]
+        ref[""] = 1
+
+    def split(self, text: str) -> List[str]:
+        """
+        Will look for the words added to the trie within `text`. Output is the original string splitted along the
+        boundaries of the words found.
+
+        This trie will match the longest possible word first !
+
+        >>> trie = Trie()
+
+        >>> trie.split("[CLS] This is a extra_id_100")
+
+        ["[CLS] This is a extra_id_100"]
+
+        >>> trie.add("[CLS]")
+
+        >>> trie.add("extra_id_1")
+
+        >>> trie.add("extra_id_100")
+
+        >>> trie.split("[CLS] This is a extra_id_100")
+
+        ["[CLS]", " This is a ", "extra_id_100"]
```

Comment:
No need for all the whitespace.
```suggestion
        >>> trie = Trie()
        >>> trie.split("[CLS] This is a extra_id_100")
        ["[CLS] This is a extra_id_100"]

        >>> trie.add("[CLS]")
        >>> trie.add("extra_id_1")
        >>> trie.add("extra_id_100")
        >>> trie.split("[CLS] This is a extra_id_100")
        ["[CLS]", " This is a ", "extra_id_100"]
```

Reply:
Actually this breaks the doc formatting tool. And fails the quality test. But indeed it would be better.

Reply:
I think this is because you did not put it in an `Example::` block.

Reply:
Make sense, done.